packages/containers/src/SidePanel/SidePanel.connect.js

```diff
@@ -1,39 +1,112 @@
+		currentCache.actions = actions;
+		currentCache.route = currentRoute;
+		currentCache.value = fn.apply();
+	}
+	return currentCache.value;
+}
+
+function isBasePathOf(pathToCheck, fullPath) {
+	const length = pathToCheck.length;
+	if (fullPath.length + 1 === length) {
+		return `${fullPath}/` === pathToCheck;
+	}
+	if (fullPath.length === length) {
+		return fullPath === pathToCheck;
+	}
+	if (fullPath.length < length) {
+		return false;
+	}
+	return pathToCheck === fullPath.slice(0, length) && fullPath[length] === '/';
+}
+
+function getActionsWrapped(actions) {
+	return actions.map(action => {
+		if (action.href && !action.onClick && !action.onClickDispatch && !action.onClickActionCreator) {
+			return {
+				...action,
+				onClick: event => {
+					event.preventDefault();
+					event.stopPropagation();
+				},
+				onClickDispatch: {
+					type: ACTION_TYPE_LINK,
+					cmf: {
+						routerPush: action.href,
+					},
+				},
+			};
+		}
+		return action;
+	});
+}
+
+function getSelectedAction(currentRoute, actions) {
+	return actions.find(action => action.href && isBasePathOf(action.href, currentRoute));
+}
+
+function getAction(id, currentRoute, state) {
+	const action = { actionId: id };
+	const info = cmf.action.getActionInfo(
+		{
+			registry: cmf.registry.getRegistry(),
+			store: {
+				getState: () => state,
+			},
+		},
+		id,
+	);
+	action.label = info.label;
+	action.id = info.id;
+
+	const route = get(
+		info,
+		'payload.cmf.routerReplace',
+		get(info, 'payload.cmf.routerPush', get(info, 'href')),
+	);
+
+	if (route && isBasePathOf(route, currentRoute)) {
+		action.active = true;
+	}
+	return action;
+}
 
 export function mapStateToProps(state, ownProps) {
 	const props = {};
+	const currentRoute = cmf.selectors.router.getPath(state);
+	const componentId = ownProps.componentId;
 	if (ownProps.actionIds) {
-		props.actions = ownProps.actionIds.map(id => {
-			const action = { actionId: id };
-			const info = cmf.action.getActionInfo(
-				{
-					registry: cmf.registry.getRegistry(),
-					store: {
-						getState: () => state,
-					},
-				},
-				id,
-			);
-			action.label = info.label;
-			action.id = info.id;
-			let route = get(info, 'payload.cmf.routerReplace');
-			if (!route) {
-				route = get(info, 'payload.cmf.routerPush');
-			}
-			if (!route) {
-				route = get(info, 'href');
-			}
-			if (route) {
-				const currentRoute = ownProps.location.pathname;
-				if (currentRoute.indexOf(route) !== -1) {
-					action.active = true;
-				}
-			}
-			return action;
-		});
+		props.actions = memoizeOne(
+			() => ownProps.actionIds.map(id => getAction(id, currentRoute, state)),
+			componentId,
+			ownProps.actionIds,
+			currentRoute,
+		);
+	} else if (ownProps.actions) {
+		props.actions = memoizeOne(
+			() => getActionsWrapped(ownProps.actions),
+			componentId,
+			ownProps.actions,
+			currentRoute,
+		);
+		props.selected = getSelectedAction(currentRoute, props.actions);
```

Comment:
no it would be a breaking change.
The plan here is to make app migrate to the href to remove this code which is in the app at the moment. CF the jira ticket